pubsub_nats.go

```diff
@@ -97,13 +97,13 @@ func (p *natsPubSub) Publish(publication *Publication, opts ...PubSubOptPublish)
 			return err
 		}
 
-		if config.requestMode == requestModeACK {
+		if config.desiredResponse == ResponseModeACK {
 			if !bytes.Equal(msg.Data, ackMessage) {
 				return fmt.Errorf("invalid ack: %s", string(msg.Data))
 			}
 		}
 
-		if config.requestMode == requestModePublication && config.responseCh != nil {
+		if config.desiredResponse == ResponseModePublication {
```

Comment:
👀 Simplified this check as it now should not be possible to enter this code path with a `nil` channel.